libraries/plugins/rc/include/steem/plugins/rc/rc_objects.hpp

```diff
@@ -74,10 +84,79 @@ class rc_account_object : public object< rc_account_object_type, rc_account_obje
       int64_t               last_max_rc = 0;
 };
 
+/**
+ * Represents a delegation pool.
+ */
+class rc_delegation_pool_object : public object< rc_delegation_pool_object_type, rc_delegation_pool_object >
+{
+   public:
+      template< typename Constructor, typename Allocator >
+      rc_delegation_pool_object( Constructor&& c, allocator< Allocator > a )
+      {
+         c( *this );
+      }
+
+      id_type                       id;
+
+      account_name_type             account;
+      steem::chain::util::manabar   rc_pool_manabar;
```

Comment:
Thinking ahead, the pool object should cache the max rc_pool_mana to avoid needing to walk the indel objects on every regeneration.

Likewise, each user should probably have a cached total outdel and indel on the rc account.